src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,54 @@
+package duke;
+
+/**
+ * This class is a parser that reads user inputs.
+ */
+public class Parser {
+
+    /**
+     * Parses the input string and calls the correct
+     * execution method.
+     * @param userInput User Input.
+     * @param taskList  TaskList.
+     * @param storage   Storage.
+     * @param textUi    TextUi.
+     */
+    public String parse(String userInput, TaskList taskList, Storage storage, TextUi textUi)
+            throws DukeException {
+        String command;
+        String body;
+
+        if (userInput.contains(" ")) {
+            command = userInput.substring(0, userInput.indexOf(" "));
+            body = userInput.substring(userInput.indexOf(" ") + 1);
+        } else {
+            command = userInput;
+            body = "";
+        }
+
+        switch (command) {
```

Comment:
Nice! You have inserted correct indentation for the `switch` and `case` statements, avoiding a commonly-made mistake.